File: Python/maxx_csv.py
# find maximum value in a list

# print out into csv
# python maxx_csv.py
# python maxx_csv.py > maxx_py_best.csv
# python maxx_csv.py | tee maxx_py_best.csv

def find_max(nums):
    return max(nums)

import time
print("n;T(n)")   # RU Excel
# print("n,T(n)") # EN Excel

for n in range(8_000_000, 40_000_001, 8_000_000):
    nums = n * [1]    # best case
#   nums = range(n)   # worst case
    for _ in range(10):
        t = time.perf_counter()
        x_max = find_max(nums)
        dt = time.perf_counter() - t
        print(f"{n};{dt:.10f}".replace(".",","), flush=True) # RU Excel
#       print(f"{n},{dt:.10f}", flush=True)                  # EN Excel
        
# (c) Valentin Arkov

File: CS/maxx_csv.cs
using System;
using System.Diagnostics;
using System.Linq;

class Program
{
    // find maximum in array (built-in)
    static int FindMax(int[] nums)
    {
        return nums.Max();
    }

    static void Main()
    {
        Console.WriteLine("n;T(n)");
        for (int n = 200_000_000; n <= 1_000_000_000; n += 200_000_000)
        {
            int[] nums = new int[n];
            for (int i = 0; i < n; i++)
                nums[i] = 1;  // Worst: i    Best: 1
            for (int i = 0; i < 10; i++)
            {
                Stopwatch stopwatch = Stopwatch.StartNew();
                int x_max = FindMax(nums);
                stopwatch.Stop();
                double t = stopwatch.Elapsed.TotalSeconds;
                Console.WriteLine($"{n};{t:F10}".Replace(',', '.'));
            }
        }
    }
}

// (c) Valentin Arkov